K2Bridge/KustoDAL/KustoResponseParser.cs

```diff
@@ -173,9 +174,13 @@ private void ReportQueryExecutionMetrics(KustoResponseDataSet kustoResponseDataS
                 response.AddParentToAgg(parent);
 
                 // Determine how to create the buckets based on the aggregation type
-                Func<DataRow, IBucket> createBucketFromDataRow = null;
+                Func<DataRow, Bucket> createBucketFromDataRow = null;
                 switch (query.PrimaryAggregation)
```

Comment:
You can simplify to 
```c#
                Func<DataRow, Bucket> createBucketFromDataRow = null;
                switch (query.PrimaryAggregation)
                {
                    case nameof(Models.Request.Aggregations.RangeAggregation):
                        createBucketFromDataRow = BucketFactory.CreateRangeBucketFromDataRow;
                        response.SetAggregationKeyed(true);
                        break;
                    case nameof(Models.Request.Aggregations.TermsAggregation):
                        createBucketFromDataRow = BucketFactory.CreateTermsBucketFromDataRow;
                        break;
                    case nameof(Models.Request.Aggregations.DateHistogramAggregation):
                        createBucketFromDataRow = BucketFactory.CreateDateHistogramBucketFromDataRow;
                        break;
                }
```
And if you prefer:
```c# 
                // Determine how to create the buckets based on the aggregation type
                Func<DataRow, Bucket> createBucketFromDataRow = query.PrimaryAggregation switch {
                    nameof(Models.Request.Aggregations.RangeAggregation) => BucketFactory.CreateRangeBucketFromDataRow,
                    nameof(Models.Request.Aggregations.TermsAggregation) => BucketFactory.CreateTermsBucketFromDataRow,
                    nameof(Models.Request.Aggregations.DateHistogramAggregation) => BucketFactory.CreateDateHistogramBucketFromDataRow,
                    _ => null,
                };
                if (query.PrimaryAggregation == nameof(Models.Request.Aggregations.RangeAggregation))
                {
                    response.SetAggregationKeyed(true);
                }
```

Reply:
Thanks! I kept the case version but with the simplified syntax.